Build failed on github workflow

Errors raised on crossbeam-utils-0.8.8.

```
     Running `rustc --crate-name crossbeam_utils --edition=2018 /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/crossbeam-utils-0.8.8/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="default"' --cfg 'feature="lazy_static"' --cfg 'feature="std"' -C metadata=c8167e716887b639 -C extra-filename=-c8167e716887b639 --out-dir /home/runner/work/ws2812-esp32-rmt-driver/ws2812-esp32-rmt-driver/target/debug/deps -L dependency=/home/runner/work/ws2812-esp32-rmt-driver/ws2812-esp32-rmt-driver/target/debug/deps --extern cfg_if=/home/runner/work/ws2812-esp32-rmt-driver/ws2812-esp32-rmt-driver/target/debug/deps/libcfg_if-f54d7254a4f5c5f1.rmeta --extern lazy_static=/home/runner/work/ws2812-esp32-rmt-driver/ws2812-esp32-rmt-driver/target/debug/deps/liblazy_static-34247ca5ec35351b.rmeta --cap-lints allow`
error[E0460]: found possibly newer version of crate `std` which `lazy_static` depends on
  --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/crossbeam-utils-0.8.8/src/sync/sharded_lock.rs:13:5
   |
13 | use lazy_static::lazy_static;
   |     ^^^^^^^^^^^
   |
   = note: perhaps that crate needs to be recompiled?
   = note: the following crate versions were found:
           crate `std`: /home/runner/.rustup/toolchains/esp/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-c5a44e5312f8d766.rlib
           crate `std`: /home/runner/.rustup/toolchains/esp/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-c5a44e5312f8d766.so
           crate `lazy_static`: /home/runner/work/ws2812-esp32-rmt-driver/ws2812-esp32-rmt-driver/target/debug/deps/liblazy_static-34247ca5ec35351b.rmeta

error: cannot find macro `lazy_static` in this scope
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/crossbeam-utils-0.8.8/src/sync/sharded_lock.rs:586:1
    |
586 | lazy_static! {
    | ^^^^^^^^^^^

error[E0425]: cannot find value `THREAD_INDICES` in this scope
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/crossbeam-utils-0.8.8/src/sync/sharded_lock.rs:604:27
    |
604 |         let mut indices = THREAD_INDICES.lock().unwrap();
    |                           ^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `THREAD_INDICES` in this scope
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/crossbeam-utils-0.8.8/src/sync/sharded_lock.rs:613:27
    |
613 |         let mut indices = THREAD_INDICES.lock().unwrap();
    |                           ^^^^^^^^^^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0425`.
error: could not compile `crossbeam-utils` due to 4 previous errors

Caused by:
  process didn't exit successfully: `rustc --crate-name crossbeam_utils --edition=2018 /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/crossbeam-utils-0.8.8/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="default"' --cfg 'feature="lazy_static"' --cfg 'feature="std"' -C metadata=c8167e716887b639 -C extra-filename=-c8167e716887b639 --out-dir /home/runner/work/ws2812-esp32-rmt-driver/ws2812-esp32-rmt-driver/target/debug/deps -L dependency=/home/runner/work/ws2812-esp32-rmt-driver/ws2812-esp32-rmt-driver/target/debug/deps --extern cfg_if=/home/runner/work/ws2812-esp32-rmt-driver/ws2812-esp32-rmt-driver/target/debug/deps/libcfg_if-f54d7254a4f5c5f1.rmeta --extern lazy_static=/home/runner/work/ws2812-esp32-rmt-driver/ws2812-esp32-rmt-driver/target/debug/deps/liblazy_static-34247ca5ec35351b.rmeta --cap-lints allow` (exit status: 1)
warning: build failed, waiting for other jobs to finish...
error: build failed
Error: Process completed with exit code 101.
```

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -19,14 +19,13 @@ jobs:
           ~/.cargo/git
           target
         key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-    - run: mv rust-toolchain.toml rust-toolchain.toml.bak
     - name: Install Rust for Xtensa
       uses: esp-rs/xtensa-toolchain@v1
       with:
         default: false
         version: "1.57.0.2"
         ldproxy: true
-    - run: mv rust-toolchain.toml.bak rust-toolchain.toml
+      working-directory: /tmp
     - name: Build
       run: cargo build --verbose
     - name: Build example m5atom
@@ -40,5 +39,4 @@ jobs:
         cargo +stable test --target x86_64-unknown-linux-gnu --lib --no-default-features --features embedded-graphics-core
     - name: Publish (Dry run)
       run: |
-        rm -rf install-rust-toolchain.sh rust-* xtensa-esp32-elf-llvm*
         cargo +stable publish --target x86_64-unknown-linux-gnu --dry-run

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -17,8 +17,9 @@ jobs:
         default: false
         version: "1.57.0.2"
         ldproxy: true
-    - uses: actions/checkout@v2
-      path: ws2812-esp32-rmt-driver
+    - uses: actions/checkout@v3
+      with:
+        path: ws2812-esp32-rmt-driver
     - uses: actions/cache@v2
       with:
         path: |

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -18,17 +18,20 @@ jobs:
         version: "1.57.0.2"
         ldproxy: true
     - uses: actions/checkout@v2
+      path: ws2812-esp32-rmt-driver
     - uses: actions/cache@v2
       with:
         path: |
           ~/.cargo/registry
           ~/.cargo/git
-          target
+          ws2812-esp32-rmt-driver/target
         key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
     - name: Build
       run: cargo build --verbose
+      working-directory: ./ws2812-esp32-rmt-driver
     - name: Build example m5atom
       run: cargo build --examples --all-features
+      working-directory: ./ws2812-esp32-rmt-driver
     - name: Test
       run: |
         cargo +stable test --target x86_64-unknown-linux-gnu --lib
@@ -36,6 +39,7 @@ jobs:
         cargo +stable test --target x86_64-unknown-linux-gnu --lib --no-default-features
         cargo +stable test --target x86_64-unknown-linux-gnu --lib --no-default-features --features smart-leds-trait
         cargo +stable test --target x86_64-unknown-linux-gnu --lib --no-default-features --features embedded-graphics-core
+      working-directory: ./ws2812-esp32-rmt-driver
     - name: Publish (Dry run)
-      run: |
-        cargo +stable publish --target x86_64-unknown-linux-gnu --dry-run
+      run: cargo +stable publish --target x86_64-unknown-linux-gnu --dry-run
+      working-directory: ./ws2812-esp32-rmt-driver

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -11,6 +11,12 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+    - name: Install Rust for Xtensa
+      uses: esp-rs/xtensa-toolchain@v1
+      with:
+        default: false
+        version: "1.57.0.2"
+        ldproxy: true
     - uses: actions/checkout@v2
     - uses: actions/cache@v2
       with:
@@ -19,13 +25,6 @@ jobs:
           ~/.cargo/git
           target
         key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-    - name: Install Rust for Xtensa
-      uses: esp-rs/xtensa-toolchain@v1
-      with:
-        default: false
-        version: "1.57.0.2"
-        ldproxy: true
-      working-directory: /tmp
     - name: Build
       run: cargo build --verbose
     - name: Build example m5atom

